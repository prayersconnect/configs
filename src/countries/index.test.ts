import cases from 'jest-in-case';
import {
  getConfigByISOName,
  getCountryAlpha2CodesByFeature,
  getCountryByISOName,
  getCountryByName,
  hasFeature,
} from './';
import prayersConfigs from './prayers-configs';

describe('country helpers', () => {
  describe('getCountryByISOName', () => {
    it('returns null if country not found', () => {
      expect(getCountryByISOName('not-found')).toBeNull();
    });

    it('returns country info if found', () => {
      expect(getCountryByISOName('United States')).toMatchSnapshot();
      expect(getCountryByISOName('Saudi Arabia')).toMatchSnapshot();
    });
  });

  describe('getCountryByName', () => {
    it('returns null if country not found', () => {
      expect(getCountryByName('not-found')).toBeNull();
    });

    it('returns country info by shortname', () => {
      expect(getCountryByName('Russia')).toMatchSnapshot();
      expect(getCountryByName('Palestine')).toMatchSnapshot();
    });

    it('returns country info by ioc', () => {
      expect(getCountryByName('USA')).toMatchSnapshot();
      expect(getCountryByName('GBR')).toMatchSnapshot();
    });
  });

  describe('getConfigByISOName', () => {
    it('returns default config if country not found', () => {
      expect(getConfigByISOName('not-found')).toMatchSnapshot();
    });

    it('returns country config for Turkey alias', () => {
      expect(getConfigByISOName('Turkey')).toEqual(
        getConfigByISOName('Turkiye')
      );
    });
    cases(
      `country configs`,
      (opts: { name: string; country: string }) => {
        expect(getConfigByISOName(opts.country)).toMatchSnapshot();
      },
      Object.keys(prayersConfigs).map((country) => {
        return {
          name: country,
          country,
        };
      })
    );
  });

  describe('hasFeature', () => {
    it('returns true if country has feature', () => {
      expect(hasFeature('United States', 'mosques')).toBe(true);
    });

    it('returns false if country does not have feature', () => {
      expect(hasFeature('Australia', 'education')).toBe(false);
    });

    it('returns false if country does not exist', () => {
      expect(hasFeature('Wonderland', 'iqamahTimes')).toBe(false);
    });
  });

  describe('getCountryAlpha2CodesByFeature', () => {
    it('returns correct countries for mosque feature', () => {
      expect(getCountryAlpha2CodesByFeature('mosques')).toMatchSnapshot();
    });
    it('returns correct countries for iqamah feature', () => {
      expect(getCountryAlpha2CodesByFeature('iqamahTimes')).toMatchSnapshot();
    });
    it('returns correct countries for events feature', () => {
      expect(getCountryAlpha2CodesByFeature('events')).toMatchSnapshot();
    });
    it('returns correct countries for education feature', () => {
      expect(getCountryAlpha2CodesByFeature('education')).toMatchSnapshot();
    });
  });
});
