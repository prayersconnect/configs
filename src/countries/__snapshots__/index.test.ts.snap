// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`country helpers getConfigByISOName country configs Australia 1`] = `
{
  "alpha2Code": "AU",
  "code": 36,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": true,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Hanafi",
    "calculation_method": "MWL",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Bangladesh 1`] = `
{
  "alpha2Code": "BD",
  "code": 50,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "bangla",
  },
  "prayerSettings": {
    "asr_method": "Hanafi",
    "calculation_method": "Karachi",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Canada 1`] = `
{
  "alpha2Code": "CA",
  "code": 124,
  "features": {
    "education": false,
    "events": true,
    "iqamahTimes": true,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "ISNA",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Egypt 1`] = `
{
  "alpha2Code": "EG",
  "code": 818,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": false,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "Egypt",
  },
}
`;

exports[`country helpers getConfigByISOName country configs France 1`] = `
{
  "alpha2Code": "FR",
  "code": 250,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "union-des-organisations-islamiques-de-france",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Germany 1`] = `
{
  "alpha2Code": "DE",
  "code": 276,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "german",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "MWL",
  },
}
`;

exports[`country helpers getConfigByISOName country configs India 1`] = `
{
  "alpha2Code": "IN",
  "code": 356,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": false,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "hindi",
  },
  "prayerSettings": {
    "asr_method": "Hanafi",
    "calculation_method": "Karachi",
  },
}
`;

exports[`country helpers getConfigByISOName country configs New Zealand 1`] = `
{
  "alpha2Code": "NZ",
  "code": 554,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": true,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Hanafi",
    "calculation_method": "MWL",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Pakistan 1`] = `
{
  "alpha2Code": "PK",
  "code": 586,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": false,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "urdu",
  },
  "prayerSettings": {
    "asr_method": "Hanafi",
    "calculation_method": "Karachi",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Russia 1`] = `
{
  "alpha2Code": "RU",
  "code": 643,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "russian",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "Russia",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Saudi Arabia 1`] = `
{
  "alpha2Code": "SA",
  "code": 682,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": false,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "Makkah",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Turkey 1`] = `
{
  "alpha2Code": "TR",
  "code": 792,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "TÃ¼rkÃ§e",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "turkey-presidency-of-religious-affairs",
  },
}
`;

exports[`country helpers getConfigByISOName country configs Turkiye 1`] = `
{
  "alpha2Code": "TR",
  "code": 792,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "TÃ¼rkÃ§e",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "turkey-presidency-of-religious-affairs",
  },
}
`;

exports[`country helpers getConfigByISOName country configs United Arab Emirates 1`] = `
{
  "alpha2Code": undefined,
  "code": 784,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": false,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "Karachi",
  },
}
`;

exports[`country helpers getConfigByISOName country configs United Kingdom 1`] = `
{
  "alpha2Code": "GB",
  "code": 826,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": true,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "mi",
    "postalCodeName": "Post Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "MWL",
  },
}
`;

exports[`country helpers getConfigByISOName country configs United States 1`] = `
{
  "alpha2Code": "US",
  "code": 840,
  "features": {
    "education": true,
    "events": true,
    "iqamahTimes": true,
    "mosques": true,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "mi",
    "postalCodeName": "ZIP Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "ISNA",
  },
}
`;

exports[`country helpers getConfigByISOName returns default config if country not found 1`] = `
{
  "alpha2Code": undefined,
  "code": null,
  "features": {
    "education": false,
    "events": false,
    "iqamahTimes": false,
    "mosques": false,
    "prayerTimes": true,
  },
  "intl": {
    "distanceUnit": "km",
    "postalCodeName": "Postal Code",
  },
  "mosque": {
    "denomination": "sunni",
    "language_services": "english",
  },
  "prayerSettings": {
    "asr_method": "Standard",
    "calculation_method": "MWL",
  },
}
`;

exports[`country helpers getCountryAlpha2CodesByFeature returns correct countries for education feature 1`] = `
[
  "US",
]
`;

exports[`country helpers getCountryAlpha2CodesByFeature returns correct countries for events feature 1`] = `
[
  "US",
  "CA",
]
`;

exports[`country helpers getCountryAlpha2CodesByFeature returns correct countries for iqamah feature 1`] = `
[
  "US",
  "CA",
  "GB",
  "AU",
  "NZ",
]
`;

exports[`country helpers getCountryAlpha2CodesByFeature returns correct countries for mosque feature 1`] = `
[
  "US",
  "CA",
  "BD",
  "TR",
  "TR",
  "GB",
  "AU",
  "NZ",
  "FR",
  "RU",
  "DE",
]
`;

exports[`country helpers getCountryByISOName returns country info if found 1`] = `
{
  "aliases": [
    "America",
    "U.S.",
    "U.S.A.",
  ],
  "alpha2": "US",
  "alpha3": "USA",
  "emoji": "ðŸ‡ºðŸ‡¸",
  "fifa": "USA",
  "ioc": "USA",
  "iso_name": "United States",
  "numeric": "840",
  "official": "United States of America",
  "short": "United States",
  "shortcode": ":flag-us:",
}
`;

exports[`country helpers getCountryByISOName returns country info if found 2`] = `
{
  "alpha2": "SA",
  "alpha3": "SAU",
  "emoji": "ðŸ‡¸ðŸ‡¦",
  "fifa": "KSA",
  "ioc": "KSA",
  "iso_name": "Saudi Arabia",
  "numeric": "682",
  "official": "Kingdom of Saudi Arabia",
  "short": "Saudi Arabia",
  "shortcode": ":flag-sa:",
}
`;
